e2e/tests/flag_test.go

```diff
@@ -94,3 +97,50 @@ func TestFlagDenyDomain(t *testing.T) {
 	newClient(t, "https://www.google.com").GET("/").ExpectStatus(http.StatusForbidden)
 	newClient(t, httpbin).GET("/status/200").ExpectStatus(http.StatusOK)
 }
+
```

Comment:
Those tests do not make any sense to me.
* There should be one setup with rx and tx
* Test must use more than one connection
*  TestFlagWriteLimit is in fact read limit
* TestFlagReadLimit is doing sth very strange

We need an httpbin endpoint that copies request body to io.Discard